firmware-management/updated/updated/source/cli.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2020 Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef UPDATED_CLI_H
+#define UPDATED_CLI_H
+
+#include <cstring>
+#include <iostream>
+#include <string>
+#include <stdexcept>
+#include <vector>
+
+#include <getopt.h>
+
+namespace updated {
+namespace cli {
+
+/**
+ * Return help text.
+ */
+const char* usage()
+{
+    return R"(Usage: updated [-l] CRITICAL|ERROR|WARNING|INFO|DEBUG|TRACE
+UpdateD, a system daemon that coordinates firmware updates.
+Example: updated -l CRITICAL
+
+Options:
+    -l          Set the logging level. Possible values: CRITICAL|ERROR|WARNING|INFO|DEBUG|TRACE
+)";
+}
+
+/**
+ * Parse command line arguments.
+ */
+std::string parse_args(const int argc, char **argv)
+{
+    if (argc < 2)
+    {
+        throw std::invalid_argument("No arguments given!");
+    }
+
+    int current_opt;
+    int optindex;
+
+    static const std::vector<option> long_opts {
+        {"log-level", required_argument, nullptr, 'l'},
```

Comment:
I think the `log-level` should be optional and we have a default (maybe `INFO`) which can be changed at compile time.